### ISSUE TITLE
Getting location info of a node

### ISSUE THREAD

**Issue by shalithasuranga** [cheeriojs/cheerio#1084]
Can we take line & column details of a specific node in cheerio? If not, is that feature possible to add with existing html parser?

Thanks

**Comment by shalithasuranga** [comment]
Any update regarding this? @matthewmueller  @jugglinmike

**Comment by trevorhreed** [comment]
If you set `xmlMode` to true, you can pass in the `withStartIndices` parameter into the `htmlparser2` instance under the hood, which in turn passes it to DomHandler. A `startIndex` property will then be attached to the node. It looks as if parse5 (the default parser for cheerio) also supports something along these lines ([see here](http://inikulin.github.io/parse5/interfaces/options.parseroptions.html)), but I'm not sure if Cheerio supports passing in the necessary parameter.

**Comment by trevorhreed** [comment]
Setting the option `xmlMode` to `true` will direct Cheerio to use htmlparser2. With that set, you can set the `withStartIndices` to `true` as well and then each node will have a `startIndex` property attached to them. However, beware that using Cheerio with `xmlMode` set to `true` may break some use cases. For example, I am using Cheerio to pull JavaScript code out of a `<script>` tag as part of a build process. With `xmlMode` set to `true`, the code gets mangled (html character encoding, etc).

After doing some digging, the new, default parser used under the hood (parse5) also has an option for adding location information to nodes (appropriately named [locationInfo](http://inikulin.github.io/parse5/interfaces/options.parseroptions.html#locationinfo)). Unfortunately, Cheerio doesn't yet support passing this option through. Perhaps, if I get some time, I'll create a PR.

**Comment by fb55** [comment]
Fixed in #1155

**Comment by pjdon** [comment]
Update since this was my first search result for this issue:

The option is now called `sourceCodeLocationInfo`. Keep in mind that with the current default parser it the locations will be in the HTML **after** cheerio is done "fixing" it.

For example

```js
const cheerio = require('cheerio');

const htmlSnippet = `\
<h2>Second Level Heading</h2>
<!-- This comment should be removed -->
<p>Some text</p>
<p>Some more text</p>
<h3>Third Level Heading</h3>
<p>Some text again</p>`;

const $ = cheerio.load(htmlSnippet, { sourceCodeLocationInfo: true });

const h3 = $('h3').get(0);
console.log(h3.startIndex, h3.endIndex);
const sliced = $.html().slice(h3.startIndex, h3.endIndex);
console.log(sliced);
```

Will yield
```
109 137
p>
<p>Some more text</p>
<h3
```
instead of 
```
114 142
<h3>Third Level Heading</h3>
```

because after loading, cheerio has already added
```
<html><head></head><body>
```
to the beginning of the string
